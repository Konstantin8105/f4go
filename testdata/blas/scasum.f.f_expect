
 REAL FUNCTION SCASUM ( N , CX , INCX )
 INTEGER INCX , N
 COMPLEX CX ( * )
 REAL STEMP
 INTEGER I , NINCX
 INTRINSIC ABS , AIMAG , REAL
 SCASUM = 0.0e0
 STEMP = 0.0e0
 IF ( N .LE. 0 .OR. INCX .LE. 0 ) RETURN
 IF ( INCX .EQ. 1 ) THEN
 DO I = 1 , N
 STEMP = STEMP + ABS ( REAL ( CX ( I ) ) ) + ABS ( AIMAG ( CX ( I ) ) )
 END DO
 ELSE
 NINCX = N * INCX
 DO I = 1 , NINCX , INCX
 STEMP = STEMP + ABS ( REAL ( CX ( I ) ) ) + ABS ( AIMAG ( CX ( I ) ) )
 END DO
 END IF
 SCASUM = STEMP
 RETURN
 END
