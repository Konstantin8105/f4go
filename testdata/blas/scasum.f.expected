COMMENT             	*> \brief \b SCASUM
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*  =========== DOCUMENTATION ===========
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	* Online html documentation available at
NEW_LINE            	

COMMENT             	*            http://www.netlib.org/lapack/explore-html/
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*  Definition:
NEW_LINE            	

COMMENT             	*  ===========
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*       REAL FUNCTION SCASUM(N,CX,INCX)
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*       .. Scalar Arguments ..
NEW_LINE            	

COMMENT             	*       INTEGER INCX,N
NEW_LINE            	

COMMENT             	*       ..
NEW_LINE            	

COMMENT             	*       .. Array Arguments ..
NEW_LINE            	

COMMENT             	*       COMPLEX CX(*)
NEW_LINE            	

COMMENT             	*       ..
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*> \par Purpose:
NEW_LINE            	

COMMENT             	*  =============
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*> \verbatim
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*>    SCASUM takes the sum of the (|Re(.)| + |Im(.)|)'s of a complex vector and
NEW_LINE            	

COMMENT             	*>    returns a single precision result.
NEW_LINE            	

COMMENT             	*> \endverbatim
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*  Arguments:
NEW_LINE            	

COMMENT             	*  ==========
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*> \param[in] N
NEW_LINE            	

COMMENT             	*> \verbatim
NEW_LINE            	

COMMENT             	*>          N is INTEGER
NEW_LINE            	

COMMENT             	*>         number of elements in input vector(s)
NEW_LINE            	

COMMENT             	*> \endverbatim
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*> \param[in,out] CX
NEW_LINE            	

COMMENT             	*> \verbatim
NEW_LINE            	

COMMENT             	*>          CX is COMPLEX array, dimension ( 1 + ( N - 1 )*abs( INCX ) )
NEW_LINE            	

COMMENT             	*> \endverbatim
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*> \param[in] INCX
NEW_LINE            	

COMMENT             	*> \verbatim
NEW_LINE            	

COMMENT             	*>          INCX is INTEGER
NEW_LINE            	

COMMENT             	*>         storage spacing between elements of SX
NEW_LINE            	

COMMENT             	*> \endverbatim
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*  Authors:
NEW_LINE            	

COMMENT             	*  ========
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*> \author Univ. of Tennessee
NEW_LINE            	

COMMENT             	*> \author Univ. of California Berkeley
NEW_LINE            	

COMMENT             	*> \author Univ. of Colorado Denver
NEW_LINE            	

COMMENT             	*> \author NAG Ltd.
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*> \date November 2017
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*> \ingroup single_blas_level1
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*> \par Further Details:
NEW_LINE            	

COMMENT             	*  =====================
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*> \verbatim
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*>     jack dongarra, linpack, 3/11/78.
NEW_LINE            	

COMMENT             	*>     modified 3/93 to return if incx .le. 0.
NEW_LINE            	

COMMENT             	*>     modified 12/3/93, array(1) declarations changed to array(*)
NEW_LINE            	

COMMENT             	*> \endverbatim
NEW_LINE            	

COMMENT             	*>
NEW_LINE            	

COMMENT             	*  =====================================================================
NEW_LINE            	

IDENT               	REAL
FUNCTION            	FUNCTION
IDENT               	SCASUM
(                   	(
IDENT               	N
,                   	,
IDENT               	CX
,                   	,
IDENT               	INCX
)                   	)
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*  -- Reference BLAS level1 routine (version 3.8.0) --
NEW_LINE            	

COMMENT             	*  -- Reference BLAS is a software package provided by Univ. of Tennessee,    --
NEW_LINE            	

COMMENT             	*  -- Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd..--
NEW_LINE            	

COMMENT             	*     November 2017
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*     .. Scalar Arguments ..
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	INCX
,                   	,
IDENT               	N
NEW_LINE            	

COMMENT             	*     ..
NEW_LINE            	

COMMENT             	*     .. Array Arguments ..
NEW_LINE            	

IDENT               	COMPLEX
IDENT               	CX
(                   	(
*                   	*
)                   	)
NEW_LINE            	

COMMENT             	*     ..
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*  =====================================================================
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*     .. Local Scalars ..
NEW_LINE            	

IDENT               	REAL
IDENT               	STEMP
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	I
,                   	,
IDENT               	NINCX
NEW_LINE            	

COMMENT             	*     ..
NEW_LINE            	

COMMENT             	*     .. Intrinsic Functions ..
NEW_LINE            	

IDENT               	INTRINSIC
IDENT               	ABS
,                   	,
IDENT               	AIMAG
,                   	,
IDENT               	REAL
NEW_LINE            	

COMMENT             	*     ..
NEW_LINE            	

IDENT               	SCASUM
=                   	=
FLOAT               	0.0e0
NEW_LINE            	

IDENT               	STEMP
=                   	=
FLOAT               	0.0e0
NEW_LINE            	

if                  	IF
(                   	(
IDENT               	N
<=                  	.LE.
INT                 	0
||                  	.OR.
IDENT               	INCX
<=                  	.LE.
INT                 	0
)                   	)
return              	RETURN
NEW_LINE            	

if                  	IF
(                   	(
IDENT               	INCX
==                  	.EQ.
INT                 	1
)                   	)
THEN                	THEN
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*        code for increment equal to 1
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

DO                  	DO
IDENT               	I
=                   	=
INT                 	1
,                   	,
IDENT               	N
NEW_LINE            	

IDENT               	STEMP
=                   	=
IDENT               	STEMP
+                   	+
IDENT               	ABS
(                   	(
IDENT               	REAL
(                   	(
IDENT               	CX
(                   	(
IDENT               	I
)                   	)
)                   	)
)                   	)
+                   	+
IDENT               	ABS
(                   	(
IDENT               	AIMAG
(                   	(
IDENT               	CX
(                   	(
IDENT               	I
)                   	)
)                   	)
)                   	)
NEW_LINE            	

END                 	END
DO                  	DO
NEW_LINE            	

else                	ELSE
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*        code for increment not equal to 1
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

IDENT               	NINCX
=                   	=
IDENT               	N
*                   	*
IDENT               	INCX
NEW_LINE            	

DO                  	DO
IDENT               	I
=                   	=
INT                 	1
,                   	,
IDENT               	NINCX
,                   	,
IDENT               	INCX
NEW_LINE            	

IDENT               	STEMP
=                   	=
IDENT               	STEMP
+                   	+
IDENT               	ABS
(                   	(
IDENT               	REAL
(                   	(
IDENT               	CX
(                   	(
IDENT               	I
)                   	)
)                   	)
)                   	)
+                   	+
IDENT               	ABS
(                   	(
IDENT               	AIMAG
(                   	(
IDENT               	CX
(                   	(
IDENT               	I
)                   	)
)                   	)
)                   	)
NEW_LINE            	

END                 	END
DO                  	DO
NEW_LINE            	

END                 	END
if                  	IF
NEW_LINE            	

IDENT               	SCASUM
=                   	=
IDENT               	STEMP
NEW_LINE            	

return              	RETURN
NEW_LINE            	

END                 	END
NEW_LINE            	

